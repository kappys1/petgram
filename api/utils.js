'use strict'

const connectDB = require('./db')
const { ObjectID } = require('mongodb')
const { errorHandler } = require('./errorHandler')

const getCollection = async (collection) => {
  try {
    const db = await connectDB()
    return await db.collection(collection)
  } catch (error) {
    errorHandler(error)
  }
}

const getAllElementsCollection = async (collection, query = {}) => await getCollection(collection).then(db => db.find(query).toArray())
const getElementCollection = async (collection, id) => await getCollection(collection).then(db => db.findOne({ _id: ObjectID(id) }))
const getElementCollectionByQuery = async (collection, query) => await getCollection(collection).then(db => db.findOne(query))
const getElementCollectionByKeyword = async (collection, keyword) => await getCollection(collection)
  .then(db => db.find({
    $text: {
      $search: keyword
    }
  }).toArray())

const createElementIntoCollection = async (collectionName, input) => {
  try {
    const collection = await getCollection(collectionName)
    const element = await collection.insertOne(input)
    input._id = element.insertedId
    return input
  } catch (error) {
    errorHandler(error)
  }
}

const editElementCollection = async (collectionName, { _id, input }) => {
  try {
    const collection = await getCollection(collectionName)
    await collection.updateOne({ _id: ObjectID(_id) }, input)
    const element = await collection.findOne({ _id: ObjectID(_id) })
    return element
  } catch (error) {
    errorHandler(error)
  }
}
const removeElementCollection = async (collectionName, { _id }) => {
  try {
    const collection = await getCollection(collectionName)
    await collection.deleteOne({ _id: ObjectID(_id) })
    const elements = await getAllElementsCollection(collectionName)
    return elements
  } catch (error) {
    errorHandler(error)
  }
}

const fillNestedTypes = async (collectionName, ids) => {
  try {
    const collection = await getCollection(collectionName)
    const elements = await collection.find({ _id: { $in: ids } }).toArray()
    return elements || []
  } catch (error) {
    errorHandler(error)
  }
}

module.exports = {
  getCollection,
  getAllElementsCollection,
  getElementCollectionByQuery,
  getElementCollection,
  removeElementCollection,
  createElementIntoCollection,
  editElementCollection,
  getElementCollectionByKeyword,
  fillNestedTypes
}
