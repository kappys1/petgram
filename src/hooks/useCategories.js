import { useEffect, useState } from 'react'

export const useCategoriesData = () => {
  const [categories, setCategories] = useState([])
  const [loading, setLoading] = useState(false)

  useEffect(function () {
    setLoading(true)
    window.fetch('https://petgram-server-kappys-5p5zyiegt-kappys1.vercel.app/categories')
      .then(res => res.json())
      .then(response => {
        setLoading(false)
        setCategories(response)
      })
  }, [])

  return { categories, loading }
}
