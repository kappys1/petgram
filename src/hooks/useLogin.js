import { useMutation } from 'react-apollo'
import { gql } from 'apollo-boost'

const LOGIN = gql`
  mutation login ($input: UserCredentials!) {
    login (input: $input)
  }
`

export const useLogin = () => {
  const [login, { data, loading, error }] = useMutation(LOGIN)
  return { login, loading, error, data }
}
