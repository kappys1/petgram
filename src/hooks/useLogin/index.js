import { useMutation } from '@apollo/client'
import { gql } from 'apollo-boost'

export const LOGIN = gql`
  mutation login ($input: UserCredentials!) {
    login (input: $input)
  }
`

export const useLogin = () => {
  const [login, { data, loading, error }] = useMutation(LOGIN)
  return { login, loading, error, data }
}
