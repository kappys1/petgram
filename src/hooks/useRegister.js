import { useMutation } from 'react-apollo'
import { gql } from 'apollo-boost'

const REGISTER = gql`
  mutation signup ($input: UserCredentials!) {
    signup (input: $input)
  }
`

export const useRegister = () => {
  const [register, { loading, error, data }] = useMutation(REGISTER)
  return { register, loading, error, data }
}
