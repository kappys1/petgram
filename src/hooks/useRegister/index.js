import { useMutation } from '@apollo/client'
import { gql } from 'apollo-boost'

export const REGISTER = gql`
  mutation signup ($input: UserCredentials!) {
    signup (input: $input)
  }
`

export const useRegister = () => {
  const [register, { loading, error, data }] = useMutation(REGISTER)
  return { register, loading, error, data }
}
