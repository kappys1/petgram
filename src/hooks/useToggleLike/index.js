import { useMutation } from '@apollo/client'
import { gql } from 'apollo-boost'

export const LIKE_PHOTO = gql`
mutation likePhoto($input: LikePhoto!) {
  likePhoto(input: $input) {
    id,
    liked,
    likes
  }
}`

export const useToggleLike = () => {
  const [toggleLike, { loading, error, data }] = useMutation(LIKE_PHOTO)
  return { toggleLike, loading, error, data }
}
