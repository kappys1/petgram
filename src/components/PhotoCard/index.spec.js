// Category.test.js
/* global describe, it, expect  */
import { render, screen } from '@testing-library/react'
import { PhotoCard } from '.'
import { LIKE_PHOTO } from '../../hooks/useToggleLike'
import { generateMock, getMockProviders } from '../../test/utils'

describe('PhotoCard', () => {
  const props = {
    id: '1',
    liked: false,
    likes: 1,
    src: ''
  }
  const mock = generateMock(LIKE_PHOTO, {})
  it('snapshot', () => {
    const mock = generateMock(LIKE_PHOTO, {})
    const Wrapper = getMockProviders(PhotoCard, mock, props)
    const { asFragment } = render(Wrapper)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render PhotoCard', async () => {
    const WrapperPhotoCard = getMockProviders(PhotoCard, mock, props)
    render(WrapperPhotoCard)
    const item = await screen.findAllByRole('article')
    expect(item).toBeTruthy()
  })
})
