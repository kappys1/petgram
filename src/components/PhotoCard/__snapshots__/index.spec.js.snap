// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoCard snapshot 1`] = `
<DocumentFragment>
  <article
    class="sc-bdfBQB ickrKr"
  />
</DocumentFragment>
`;
